website/preprints/tasks.py

```diff
@@ -30,22 +32,66 @@ def on_preprint_updated(preprint_id, update_share=True, old_subjects=None):
         except HTTPError as err:
             sentry.log_exception()
             sentry.log_message(err.args[0])
+    if update_share:
+        update_preprint_share(preprint, old_subjects)
 
-    if settings.SHARE_URL and update_share:
+def update_preprint_share(preprint, old_subjects=None):
+    if settings.SHARE_URL:
         if not preprint.provider.access_token:
             raise ValueError('No access_token for {}. Unable to send {} to SHARE.'.format(preprint.provider, preprint))
-        resp = requests.post('{}api/v2/normalizeddata/'.format(settings.SHARE_URL), json={
-            'data': {
-                'type': 'NormalizedData',
-                'attributes': {
-                    'tasks': [],
-                    'raw': None,
-                    'data': {'@graph': format_preprint(preprint, old_subjects)}
-                }
-            }
-        }, headers={'Authorization': 'Bearer {}'.format(preprint.provider.access_token), 'Content-Type': 'application/vnd.api+json'})
-        logger.debug(resp.content)
+        _update_preprint_share(preprint, old_subjects)
+
+def _update_preprint_share(preprint, old_subjects):
+    data = serialize_share_preprint_data(preprint, old_subjects)
+    resp = send_share_preprint_data(preprint, data)
+    try:
+        assert False
+        resp.raise_for_status()
+    # any exception gets sent to deskk
+    except Exception:
+        if resp.status_code >= 500:
+            _async_update_preprint_share.delay(preprint._id, old_subjects)
+        else:
+            send_desk_share_preprint_error(preprint, resp, 0)
+
+@celery_app.task(bind=True, max_retries=4, acks_late=True)
+def _async_update_preprint_share(self, preprint_id, old_subjects):
+    # Any modifications to this function may need to change _async_update_share
```

Comment:
@brianjgeiger More a defensive/paranoia thing. I think I separated the  _update and _async_update functions well enough that any changes should only have to occur to the functions both call. 

A few minutes thought and I couldn't come up with a good way to guard against them changing.